meson.build

```diff
@@ -44,9 +44,28 @@ elif bpf_clang_maj < 17
           .format(bpf_clang.full_path(), bpf_clang_ver))
 endif
 
-if get_option('libbpf_a') != ''
+libbpf_a = '@0@/libbpf/src/libbpf.a'.format(meson.current_source_dir())
+
+if get_option('libbpf_a') == 'disabled'
+  libbpf_a = ''
+elif get_option('libbpf_a') != ''
+  libbpf_a = get_option('libbpf_a')
+endif
+
+if libbpf_a != ''
+  if not fs.exists(libbpf_a)
+    # If the user isn't providing their own libbpf_a (use the submodule)
+    if get_option('libbpf_a') == ''
+      message('Checking out libbpf submodule and building it...')
+      run_command('git', 'submodule', 'update', '--init', '--recursive', check: true)
+      run_command('make', '-C', 'libbpf/src', check: true)
```

Comment:
Shouldn't it pass `cc` and other build options (`CFLGAS`, `LDFLAGS`) down to `make`? Also, what about build concurrency? Can we specify a reasonable `-j` option by default?

Reply:
Doesn't cc (and the other build options) get picked up by make in the same way that meson grabs it with `meson.get_compiler`? When I'm building locally if I run setup with the `env CC=clang` then it builds libbpf with clang as well (as opposed to gcc which is my default).

Reply:
> Can we specify a reasonable -j option by default?

Sure. Is there a decent middle ground here? `-j32` ?

Reply:
> Doesn't cc (and the other build options) get picked up by make in the same way that meson grabs it with `meson.get_compiler`? When I'm building locally if I run setup with the `env CC=clang` then it builds libbpf with clang as well (as opposed to gcc which is my default).

Well, depends on how the flags are specified, I think. e.g. I don't think CFLAGS is the only way to pass in compiler flags to meson. Best to explicitly set relavent enviroments as when we're calling e.g. cargo.

Reply:
> > Can we specify a reasonable -j option by default?
> 
> Sure. Is there a decent middle ground here? `-j32` ?

Add `nproc` as dependency and run that to get the answer?

Reply:
>  e.g. I don't think CFLAGS is the only way to pass in compiler flags to meson

In looking at the docs, I think this is the recommended way to do this: https://mesonbuild.com/howtox.html#set-extra-compiler-and-linker-flags-from-the-outside-when-eg-building-distro-packages

I think if there are specific flags that we want to pass to either clang or gcc then we have to do that explicitly in the build file. Maybe I'm missing something.